Packs/Mimecast/Integrations/MimecastEventCollector/MimecastEventCollector.py

```diff
@@ -0,0 +1,567 @@
+        """
+        self.options.limit = 1 if demisto.command() == 'test-module' else SIEM_LOG_LIMIT
+        stored = []
+        if self.events_from_prev_run and self.options.limit:
+            if len(self.events_from_prev_run) >= self.options.limit:
+                demisto.info(
+                    f'{self.options.limit=} reached. \
+                    slicing from {len(self.events_from_prev_run)=}.'
+                )
+                stored = self.events_from_prev_run[:self.options.limit]
+                self.events_from_prev_run = stored[self.options.limit:]
+                return stored
+
+            else:
+                stored = self.events_from_prev_run
+                demisto.info(f'added to stored {self.events_from_prev_run=}')
+                self.events_from_prev_run = []
+
+        for logs in self._iter_events():
+            stored.extend(logs)
+            if self.options.limit and len(stored) >= self.options.limit:
+                demisto.info(
+                    f'{self.options.limit=} reached. \
+                    slicing from {len(logs)=}.'
+                )
+                self.events_from_prev_run = stored[self.options.limit:]
+                demisto.info(f'storing {len(self.events_from_prev_run)} siem events for next run')
+                return stored[: self.options.limit]
+
+        return stored
+
+    def _iter_events(self):  # pragma: no cover
+        self.client.request = IntegrationHTTPRequest(**(self.get_req_object_siem()))
+        response = self.call()
+        events = self.process_siem_response(response)
+        if not events:
+            return []
+
+        while True:
+            demisto.info(f'\n {len(events)} Siem logs were fetched from mimecast \n')
+            yield events
+
+            self.client.request = IntegrationHTTPRequest(**(self.get_req_object_siem()))
+            response = self.call()
+            events = self.process_siem_response(response)
+            if not events:
+                break
+
+    def process_siem_response(self, response):  # ignore: type
+        """
+        Args:
+            response (Request.Response) - The response from the mimecast API
+
+        Returns:
+            The events after process and modification.
+        """
+        resp_body = response.content
+        resp_headers = response.headers
+        content_type = resp_headers['Content-Type']
+
+        # End if response is JSON as there is no log file to download
+        if content_type == 'application/json':
+            decoded_response = resp_body.decode('utf8')
+            json_response = json.loads(decoded_response)
+
+            if fail_reason := json_response.get('fail', []):
+                return_error(f'There was an error with siem events call {fail_reason}')
+            demisto.info('No more logs available')
+            return []
+        # Process log file
+        elif content_type == 'application/octet-stream':
```

Comment:
This `content_type` check should be case insensitive.

Reply:
Done